pkg/kube/inject/initializer.go

```diff
@@ -21,26 +21,8 @@ import (
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-
-	"istio.io/istio/pkg/config/constants"
-	"istio.io/istio/pkg/util/sets"
 )
 
-// IgnoredNamespaces contains the system namespaces referenced from Kubernetes:
-// Ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#viewing-namespaces
-// "kube-system": The namespace for objects created by the Kubernetes system.
-// "kube-public": This namespace is mostly reserved for cluster usage.
-// "kube-node-lease": This namespace for the lease objects associated with each node
-// which improves the performance of the node heartbeats as the cluster scales.
-// "local-path-storage": Dynamically provisioning persistent local storage with Kubernetes.
-//
-//	used with Kind cluster: https://github.com/rancher/local-path-provisioner
-var IgnoredNamespaces = sets.New(
```

Comment:
On one hand, we are skipping over https://github.com/istio/istio/blob/fe53894c79e7bbd0b30f42040990f7d7e79c5b89/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml#L144.

However, if we were to make this respect the new config, openshift's config to  allow kube-system would cause an outage, as it places a circular dependency on kube-system and istio.

Seems like maybe we don't want this to be "injection" at all and only tie it to the namespace ca cert?

Reply:
agree, I don't think more to injection options are ideal as we move to ambient ( and injection-less), and mixing CA and injection features doesn't make sense.